annotations/views.py

```diff
@@ -685,16 +685,43 @@ def add_text_to_collection(request, *args, **kwargs):
     return JsonResponse({})
 
 
-@login_required
 def user_details(request, userid, *args, **kwargs):
+    """
+
+    :param request: The user request to fetch user details
+    :type: Request()
+    :param userid:
+    :type int
+    :param args:
+    :type list
+    :param kwargs:
+    :type dict
+    :return: HttpResponse
+    """
     user = get_object_or_404(VogonUser, pk=userid)
-    template = loader.get_template('annotations/user_details.html')
-    context = RequestContext(request, {
-        'user': request.user,
-        'detail_user': user,
-    })
-    return HttpResponse(template.render(context))
+    if request.user.is_authenticated():
```

Comment:
We have modified existing view which required login to display user details i.e his own profile.
In case the user is not loggedIn the public profile page is displayed which is a new HTML page